WooCommerce/WooCommerceTests/POS/ViewModels/ItemListViewModelTests.swift

```diff
@@ -211,21 +224,38 @@ final class ItemListViewModelTests: XCTestCase {
         // When
         await sut.populatePointOfSaleItems()
 
+        sut.$shouldShowHeaderBanner
+            .dropFirst()
+            .removeDuplicates()
+            .sink { value in
+            XCTAssertEqual(value, false)
+            expectation.fulfill()
+        }
+        .store(in: &cancellables)
+
         // Then
         XCTAssertEqual(sut.state, .error(expectedError))
-        XCTAssertEqual(sut.shouldShowHeaderBanner, false)
     }
 
     func test_shouldShowHeaderBanner_when_itemListViewModel_loaded_then_returns_true() async {
         // Given
+        let expectation = XCTestExpectation(description: "Banner should be shown once products are loaded")
         let expectedItems = Self.makeItems()
 
         // When
         await sut.populatePointOfSaleItems()
 
+        sut.$shouldShowHeaderBanner
+            .dropFirst()
+            .removeDuplicates()
+            .sink { value in
+            XCTAssertEqual(value, true)
+            expectation.fulfill()
+        }
+        .store(in: &cancellables)
+
         // Then
         XCTAssertEqual(sut.state, .loaded(expectedItems))
```

Comment:
This assertion is not needed for this test. Try the last assertion to be:

`XCTAssertEqual(sut.shouldShowHeaderBanner, true)` or just `await fulfillment(of: [expectation], timeout: 1.0)`